### PR TITLE
gather some test utilities under `xaynet_core::testutils`

### PR DESCRIPTION
**Pull request by little-dude**
This makes some helpers for crafting messages re-usable from other
crates. `xaynet_core::testutils` requires the "testutils" feature.

This commit also moves the benchmarks to a separate crate.

### CODE REVIEW

File: rust/benches/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Robert-Steiner) [inline rust/benches/src/lib.rs:1]:
```suggestion

```

File: rust/xaynet-client/Cargo.toml [hunk 2]
```diff
@@ -19,13 +19,14 @@ bincode = "1.3.1"
 thiserror = "1.0.20"
 tracing = "0.1.19"
 async-trait = "0.1.40"
-xaynet-core = { path = "../xaynet-core", version = "0.1.0" }
+xaynet-core = { path = "../xaynet-core", version = "0.1.0", features = [] }
```
Comment (Robert-Steiner) [inline rust/xaynet-client/Cargo.toml:22]: Do we need to add `features` here?

File: rust/benches/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "benches"
+version = "0.0.0"
+authors = ["Xayn Engineering <engineering@xaynet.dev>"]
+edition = "2018"
+publish = false
+
+[dependencies]
+criterion = "0.3.3"
+xaynet-core = { version = "0.1.0", path = "../xaynet-core", features = ["testutils"] }
```
Comment (Robert-Steiner) [inline rust/benches/Cargo.toml:10]: Not sure but do we need to add 
```toml
[[bench]]
name = "messages"
```
here?

**Review by Robert-Steiner — APPROVED** [review]
The feature flag is a bit hacky, but I don't have a better solution 😅 However, we only add the flag in one cfg annotation, so I'm totally fine with that. Like the idea of the benches crate 👍

**Comment by little-dude** [comment]
@Robert-Steiner thanks for the review (and for spotting that the benchmarks were actually broken :confounded: ). I think I addressed all of your comments. Let's see if the CI is happy.

**Review by janpetschexain — APPROVED** [review]
looks good. the tuple outputs of the helper functions are sometimes a bit hard to read when used, one needs to jump back and forth between definition and call. sth like a struct output with named fields would be nice to read, but this would also introduce a lot of additional code. those helpers are anyways only meant for internal usage so i guess that's ok to have tuples.

**Review by finiteprods — COMMENTED** [review]
if I'm understanding the compromise here: although an update message helper feels like it ought to live with the update message module (say), the fact is it's used for tests in multiple places. so it's justifiable for such helpers to live in a shared crate (and necessary even, to be accessible from a different crate like benchmarks). And we put the helpers behind a feature flag because we consider them "internal".

**Comment by little-dude** [comment]
> if I'm understanding the compromise here: although an update message helper feels like it ought to live with the update message module (say), the fact is it's used for tests in multiple places. so it's justifiable for such helpers to live in a shared crate (and necessary even, to be accessible from a different crate like benchmarks). And we put the helpers behind a feature flag because we consider them "internal".

Exactly

**Pull request merged**